hr_applicant/models/training.py

```diff
@@ -0,0 +1,289 @@
+    job_id = fields.Many2one("hr.job", "Applied Job")
+    department = fields.Char(related="job_id.name", string="Department", readonly=True)
+    training_location = fields.Char("Training Location")
+    duration = fields.Integer("Course Duration", required=True)
+    duration_type = fields.Selection(
+        [("day", "Days"), ("week", "Weeks"), ("month", "Months")], required=True
+    )
+    local_short_description = fields.Text("Course Short Description")
+
+
+class TrainingClass(models.Model):
+
+    _name = "training.class"
+    _rec_name = "course_id"
+    _description = "Training Class"
+
+    @api.constrains("training_start_date", "training_end_date")
+    def _check_training_dup(self):
+        self.ensure_one()
+        if self.training_start_date < fields.Date.context_today(self):
+            raise ValidationError(_("You can't create past training!"))
+        if self.training_start_date > self.training_end_date:
+            raise ValidationError(
+                _("End Date should be greated than Start date of Training!")
+            )
+
+    course_id = fields.Many2one("training.courses", string="Course Name", required=True)
+    department = fields.Char(
+        related="course_id.department", string="Department", readonly=True
+    )
+    job_id = fields.Many2one(
+        related="course_id.job_id",
+        comodel="hr.job",
+        string="Applied Job",
+        readonly=True,
+    )
+    course_categ_id = fields.Many2one(
+        related="course_id.course_type_id",
+        comodel="course.type",
+        string="Course Type",
+        readonly=True,
+    )
+    training_location = fields.Char(
+        related="course_id.training_location", string="Training Location", readonly=True
+    )
+    training_attendees = fields.Integer("Training Attendees", required=True)
+    training_start_date = fields.Date(
+        "Training Start Date",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        required=True,
+    )
+    training_end_date = fields.Date(
+        "Training End Date", readonly=True, states={"draft": [("readonly", False)]}
+    )
+    attendees_ids = fields.One2many(
+        "list.of.attendees", "class_id", string="List of Local Attendees"
+    )
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("to_be_approved", "To Be Approved"),
+            ("approved", "Approved"),
+            ("completed", "Completed"),
+            ("cancel", "Cancelled"),
+        ],
+        "State",
+        default="draft",
+    )
+    description = fields.Text("Description")
+
+    @api.onchange("training_start_date", "course_id")
+    def _onchange_start_date(self):
+        for rec in self:
+            if rec.training_start_date and rec.course_id:
+                end_date = False
+                if rec.course_id.duration and rec.course_id.duration_type == "day":
+                    end_date = rec.training_start_date + datetime.timedelta(
```

Comment:
Please check the wronlgy use timedelta. It should import directly from the main datetime library, not from the sub class.
Import Like : 
          from datetime import datetime, timedelta

And then use timedelta. 
Also You have already used the relativedelta , so better we use relativedelta and remove to call additional timedelta.